src/wrapper_array.jl

```diff
@@ -155,21 +155,16 @@ for f in (:getindex, :view, :dotview)
             return Base.$f(A, inds...)
         end
 
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, inds::Vararg{<:Integer})
-            # Easy scalar case, will just return the element
-            return Base.$f(parent(a), inds...)
-        end
-
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, ci::CartesianIndex)
-            # Easy scalar case, will just return the element
-            return Base.$f(parent(a), ci)
-        end
-
-        @propagate_inbounds function Base.$f(a::NamedDimsArray, inds...)
-            # Some nonscalar case, will return an array, so need to give that names.
+        @propagate_inbounds function Base.$f(a::NamedDimsArray, raw_inds...)
+            inds = Base.to_indices(a, raw_inds)
             data = Base.$f(parent(a), inds...)
```

Comment:
It is tempting here to insert
```julia
data isa AbstractArray || return data
```
and thus only run `remaining_dimnames_from_indexing` at all when the answer is not a scalar. Although it should be compiled away, I think, and this goes back to having two paths which return `data`. 

BTW, the speed of indexing like `@btime $nda[1,1]` is unaffected by this PR.